src/main.ts

```diff
@@ -454,4 +445,33 @@ async function main() {
 	);
 }
 
-main().catch(console.log);
+async function getHashForBlock(
+	api: ApiPromise,
+	blockId: string
+): Promise<BlockHash> {
+	try {
+		const isHexStr = isHex(blockId);
+		if (isHexStr && blockId.length === 66) {
+			return api.createType('BlockHash', blockId);
+		} else if (isHexStr) {
+			throw {
+				error: `Cannot get block hash for ${blockId}. Hex string block IDs must be 66-characters (32-bytes) in length.`,
```

Comment:
I think it'd be good to convert this to a number if some people's systems use hex. There's no way block number collides with a hash that's 32 bytes.

Reply:
Not sure I totally understand, but would [the conversion](https://github.com/paritytech/substrate-api-sidecar/blob/master/src/utils.ts#L8) be surfacing an additional point of failure/ misdirection that could obscure the original error? i.e if someone accidentally sends the string "Hello", but we convert it to a number before interpolating into the error message, it could throw off debugging.

Reply:
Yeah, I think I'm with Zeke on this one but I'm still not completely sure this is what Joe was suggesting so I'll just hold off.